### DISCUSSION TITLE
Add star ( * ) to required field names

### DISCUSSION THREAD

**Discussion by BennyAlex** [lookfirst/mui-rff#1025] [General]
In MUI as well in material design in gerneral, required field are indicated by a `*` at the end of the name.
It would be helpfull, if this would be added automatically to any field marked as required in yup.

**Comment by lookfirst** [comment]
Hi there, thanks for the comment. I've moved this to a discussion.

Have you tried this method? https://github.com/lookfirst/mui-rff#makerequiredschema

**Reply by BennyAlex** [reply]
Yeah I use the makeRequired function but this doesn't add a star to the label.
When setting required={true} it does work.

**Comment by BennyAlex** [comment]
The problem is, that all fields in the required object are false:
![image](https://user-images.githubusercontent.com/17402724/225149176-0680002f-c8fe-4826-b77b-8d6da5cce38f.png)

**Reply by lookfirst** [reply]
When you log out required at that point in the execution of your app, why would you expect them to be anything other than false?

**Reply by BennyAlex** [reply]
since MUI expects true if you want to mark a field as required.
Not sure what you want to track with the required object otherwise...

So if a field has yup.required() it should return true, otherwise false.

Required is not a dynamic validation, its a fixed boolean

**Reply by BennyAlex** [reply]
@lookfirst 
Hey, can you give me a explanation, why they should be all false, please? :D

**Reply by lookfirst** [reply]
Can you show me your schema? Maybe you've declared it incorrectly?

**Reply by BennyAlex** [reply]
Is not working in the demo codesandbox, I changed your code a bit to show the problem:
https://codesandbox.io/s/react-final-form-material-ui-example-forked-jq8nui?file=/src/index.tsx

![image](https://user-images.githubusercontent.com/17402724/225981064-5bd90d9f-b1f4-44d5-a38e-c7f1b1de55b8.png)

**Reply by BennyAlex** [reply]
@lookfirst  if you look at the console, you see, every required prop is false

**Reply by lookfirst** [reply]
Ah, that codesandbox is using a really really old version of mui-rff, and as we all know, javascript code rots over time. You have to stay up to date with the latest of things for anything to work properly.

What I did was run the example in the latest version (cd example; yarn; yarn start), and added the console.log in the same place that you did in the codesandbox... as you can see, everything works just fine.

I hope this helps you.

![2023-03-17 at 19 31 13 png](https://user-images.githubusercontent.com/85355/226002631-fba6dba0-0390-42b0-8230-4cc354ca6440.png)

**Reply by BennyAlex** [reply]
@lookfirst 
Hey, thanks for have a look. However, I cant get it to work, neither locally or on codesandbox.
See here please: https://codesandbox.io/s/pedantic-snow-czymrw

**Reply by lookfirst** [reply]
I was using a field in Yup schema that wasn't exposed in order to get that required data. It was a bit hacky because Yup didn't make this information as part of their public api.  I even filed an issue years ago, but got no love. https://github.com/jquense/yup/issues/622

It seems that Yup finally broke it in 1.x.

Pin the dependency to 0.32.x (which is the dep in mui-rff) and things start to work again.

Now, the question going forward... how to fix this in Yup? I don't know... what I did before was set a breakpoint and then look deeply into the schema object to find what I wanted. Maybe they just moved things around or maybe they just made things really difficult.

I've been also thinking that switching to [zod](https://github.com/lookfirst/mui-rff/issues/885) might also be a good idea, but I haven't found the motivation yet and I hate to break people's code.

**Reply by BennyAlex** [reply]
yeah i thought also that it has to do with yup. You should make it an optional peer dependency with yup <= 0.32

**Reply by lookfirst** [reply]
![2023-03-20 at 01 05 39 png](https://user-images.githubusercontent.com/85355/226223161-849d769c-11e7-4705-863e-c39211641d7f.png)
